packages/core/datadog-sourcemaps.gradle

```diff
@@ -0,0 +1,139 @@
+import org.apache.tools.ant.taskdefs.condition.Os
+
+def reactConfig = project.hasProperty("react") ? project.react : [:]
+def reactRoot = file(reactConfig.root ?: "../../")
+
+def datadogConfig = new Properties()
+def datadogConfigFile = new File(reactRoot, "datadog-sourcemaps.properties")
+
+if (datadogConfigFile.exists()) {
+    datadogConfigFile.withInputStream {
+        datadogConfig.load(it)
+    }
+}
+
+afterEvaluate {
+
+    def androidExtension = extensions.findByName("android")
+
+    if (androidExtension == null) {
+        throw new GradleException("Android extension is missing.")
+    }
+
+    // TODO service name per variant?
+    def serviceName = datadogConfig.SERVICE_NAME ?: androidExtension.defaultConfig.applicationId
+
+    if (serviceName == null) {
+        throw new GradleException("Cannot determine service name.")
+    }
+
+    def releaseVersion = androidExtension.defaultConfig.versionName
+    def buildVersion = androidExtension.defaultConfig.versionCode
+
+    if (releaseVersion == null) {
+        throw new GradleException("Cannot determine application version," +
+                " probably not running in the app module.")
+    }
+
+    androidExtension.applicationVariants.all { variant ->
+        // locations and names are defined in "../../node_modules/react-native/react.gradle"
+        def targetName = variant.name.capitalize()
+        def targetPath = variant.dirName
+
+        def bundleTask = tasks.findByName("bundle${targetName}JsAndAssets")
```

Comment:
Question: does this make us rely too much on the RN `react.gradle` implementation?

Reply:
That is likely to break in a new version change, and in the meantime new projects won't be able to use our SDK, and existing projects won't be able to upgrade their RN versions.
We can mitigate it by testing new releases and watching for changes in the `react.gradle` file.

My point is that we should be careful when relying on RN implementation, so that's why I'm asking the question to make sure we do not miss any opportunity.

For this one I think we could have done a regex to be a bit more flexible, but that does not reduce the risk of bug by a lot, so I'm ok with this implementation

Reply:
imo we have to rely on the implementation anyway, because tasks added by RN SDK don't have any public API or guarantees. Same for the bundling tool.

So either way we can get an access to the bundles/sourcemaps only by relying on the implementation.